app/controllers/AdditionalMetadataClosureStatusController.scala

```diff
@@ -0,0 +1,85 @@
+                                                          val keycloakConfiguration: KeycloakConfiguration,
+                                                          val controllerComponents: SecurityComponents,
+                                                          val cache: AsyncCacheApi
+                                                         ) extends TokenSecurity {
+
+
+  val closureStatusForm: Form[ClosureStatusFormData] = Form(
+    mapping(
+      "closureStatus" -> boolean
+        .verifying("You must confirm this closure has been approved before continuing.", b => b)
+    )(ClosureStatusFormData.apply)(ClosureStatusFormData.unapply)
+  )
+
+  val closureStatusField: InputNameAndValue = InputNameAndValue("closureStatus", "Yes, I confirm")
+
+
+  def getClosureStatusPage(consignmentId: UUID, fileIds: List[UUID]): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+
+    val filters = Option(FileFilters(None, Option(fileIds), None))
+
+    for {
+      details <- consignmentService.getConsignmentDetails(consignmentId, request.token.bearerAccessToken)
+      consignment <- consignmentService.getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, filters)
+      response <- if (consignment.files.nonEmpty) {
+        val filePaths = consignment.files.flatMap(_.fileMetadata).filter(_.name == clientSideOriginalFilepath).map(_.value)
+        val closureStatus = consignment.files.flatMap(_.fileMetadata).find(_.name == closureType.name).exists(_.value == closureType.value)
+        cache.set(s"$consignmentId-data", (consignment.consignmentReference, filePaths, details.parentFolderId.get), 1.hour)
+        Future(Ok(views.html.standard.additionalMetadataClosureStatus(consignmentId, filePaths, fileIds, closureStatusForm, closureStatusField, closureStatus, consignment.consignmentReference,
+          details.parentFolderId.get, request.token.name)))
+      } else {
+        Future.failed(new IllegalStateException(s"Can't find selected files for the consignment $consignmentId"))
+      }
+    } yield response
+  }
+
+
+  def submitClosureStatus(consignmentId: UUID, fileIds: List[UUID]): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+
+    val errorFunction: Form[ClosureStatusFormData] => Future[Result] = { formWithErrors: Form[ClosureStatusFormData] =>
+      for {
+        (consignmentRef, filePaths, parentFolderId) <- cache.getOrElseUpdate[(String, List[String], UUID)](s"$consignmentId-data", 1.hour)(
+          for {
+            details <- consignmentService.getConsignmentDetails(consignmentId, request.token.bearerAccessToken)
+            consignment <- consignmentService.getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, Option(FileFilters(None, Option(fileIds), None)))
+            filePaths = consignment.files.flatMap(_.fileMetadata).filter(_.name == clientSideOriginalFilepath).map(_.value)
```

Comment:
Why are the "file paths" needed?

Reply:
It's required because we are showing the name of the file as per UI design https://tdr-prototype.herokuapp.com/metadata/closure-metadata/T2-flour/check-closure-status